src/app/components/Field/Field.test.tsx

```diff
@@ -0,0 +1,196 @@
+import { render, screen, waitFor } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+
+import { Factory } from '../../../game/services/Factory'
+import { updateField } from '../../../game/reducers/update-field'
+import { stubGame } from '../../../test-utils/stubs/game'
+import { carrot, pumpkin } from '../../../game/cards'
+import * as cards from '../../../game/cards'
+import { isCardId } from '../../../game/types/guards'
+import { playedCropWrapperClassName } from '../PlayedCrop'
+import { cardClassName } from '../Card/CardTemplate'
+
+import { Field, FieldProps, rotationTransform } from '.'
+
+let gameStub = stubGame()
+const opponentPlayerId = Object.keys(gameStub.table.players)[1]
+
+const fieldCrop1 = carrot
+const fieldCrop2 = pumpkin
+
+const cropsStub = [
+  { ...Factory.buildPlayedCrop(fieldCrop1), waterCards: 1 },
+  { ...Factory.buildPlayedCrop(fieldCrop2), waterCards: 3 },
+]
+
+gameStub = updateField(gameStub, gameStub.sessionOwnerPlayerId, {
+  crops: cropsStub,
+})
+
+gameStub = updateField(gameStub, opponentPlayerId, {
+  crops: cropsStub,
+})
+
+const StubField = (overrides: Partial<FieldProps>) => {
+  return (
+    <Field
+      game={gameStub}
+      playerId={gameStub.sessionOwnerPlayerId}
+      {...overrides}
+    />
+  )
+}
+
+describe('Field', () => {
+  test('renders crop cards for self', () => {
+    render(<StubField />)
+
+    for (const crop of cropsStub) {
+      if (!isCardId(crop.id)) {
+        throw new Error()
+      }
+
+      const card = screen
+        .getByText(cards[crop.id].name)
+        .closest(`.${cardClassName}`)
+
+      const { transform } = getComputedStyle(card!)
+      expect(transform).toMatchSnapshot()
```

Comment:
i'm having mixed feelings about expectations against the computed styles in here. i get that some of these tests are testing visual things, and not functional things, so there's not really any other way to test those in a unit test, but for something like this one what's the benefit of checking transform vs just selecting by text?

Reply:
I spent some time looking into visual regression testing options (BackstopJS and jest-image-snapshot, specifically). These are very powerful tools, but they are quite heavy and seem excessive given the narrow scope of our current needs. I think it's most pragmatic to stick with the current snapshot-based approach and reevaluate if it becomes clear that we have a more common need for visual regression testing. What do you think, @lstebner?

Reply:
yeah, that sounds good. i've always wanted to put visual regression tests to good use, but i honestly have always landed on it not being ideal. thanks for investigating them though!

i don't think you need to make any changes for this PR, but just as i was typing this response it occurred to me what would probably be ideal is having some utility functions for these transforms. that way it isn't hidden away in these other components, but they could instead do something like `getFocusTransform(event.target.getBoundingClientRect())` and then test wise the utility can be tested, and the component test could use a mock to make sure it uses the appropriate helper at the appropriate time. sorry i didn't think of this sooner to mention it, but i'm also good with how you have things now. if these transforms eventually get repeated maybe we can consider these utilities for it instead.